internal/timespec/timespec.go

```diff
@@ -28,3 +28,12 @@ func CStructToTimespec(cts CTimespecPtr) Timespec {
 		Nsec: int64(t.tv_nsec),
 	}
 }
+
+// CopyToCStruct copies the time values from a Timespec to a previously
+// allocated C `struct timespec`. Due to restrictions on Cgo the C pointer
+// must be passed via the CTimespecPtr wrapper.
+func CopyToCStruct(ts Timespec, cts CTimespecPtr) {
```

Comment:
Could you elaborate these "restrictions" so I'm sure I understand correctly?

Reply:
cgo doesn't let you make public functions that reference a C.type type because its technically not exported. Even though this module is "internal" and not meant for public consumption it won't even build.
The convention I created for internal/<blah> packages was to make specific types over unsafe.Pointer to allow the functions to be exported but still acting as a reminder what underlying C type the unsafe.Pointer was meant to represent.

Reply:
Ok, I see. So, these `unsafe.Pointer` aliases, can you confirm, that their usage is in accordance with the [rules](https://golang.org/cmd/cgo/#hdr-Passing_pointers)? If not, I would double check that for you. I would like to make sure, that we don't get issues, when the GC kicks in.

Reply:
I'm pretty sure, but it would be totally fine if you want to double check. Since the pointer is to C memory I don't think the Go GC will mess with it, and it is a legit pointer. The type aliasing shouldn't have an impact on the behavior IMO.

Reply:
Yeah, since you don't introduce that here, I might check later, but not for the approval of this PR.